ETANotifications/Services/NotificationService.cs

```diff
@@ -10,25 +13,30 @@ public interface INotificationService
 
     public class NotificationService : INotificationService
     {
-        private readonly TwilioRestClient _twilioRestClient;
-
+        private readonly ITwilioRestClient _client;
         private readonly string _accountSid = WebConfigurationManager.AppSettings["TwilioAccountSid"];
         private readonly string _authToken = WebConfigurationManager.AppSettings["TwilioAuthToken"];
         private readonly string _twilioNumber = WebConfigurationManager.AppSettings["TwilioPhoneNumber"];
 
         public NotificationService()
         {
-            _twilioRestClient = new TwilioRestClient(_accountSid, _authToken);
+            _client = new TwilioRestClient(_accountSid, _authToken);
         }
 
-        public NotificationService(TwilioRestClient twilioRestClient)
+        public NotificationService(ITwilioRestClient client)
         {
-            _twilioRestClient = twilioRestClient;
+            _client = client;
         }
 
         public void SendSmsNotification(string phoneNumber, string message, string statusCallback)
         {
-            _twilioRestClient.SendMessage(_twilioNumber, phoneNumber, message, statusCallback);
+            var to = new PhoneNumber(phoneNumber);
+            MessageResource.Create(
```

Comment:
The tutorial freezes at this point because of a bug we found last week on synchronous methods on the twilio-csharp library. As a workaround we should use the asynchronous version in multithreaded cases like so: 
```csharp
public async Task SendSmsNotification(string phoneNumber, string message, string statusCallback)
        {
            var to = new PhoneNumber(phoneNumber);
            await MessageResource.CreateAsync(
                to,
                from: new PhoneNumber(_twilioNumber),
                body: message,
                statusCallback: new Uri(statusCallback),
                client: _client);
        }
```
I also recommend executing the tutorial to ensure is free of errors.

Bare in mind that other methods calling this one will now have to `await` for it to finish.